app/Compile.hs

```diff
@@ -21,65 +22,83 @@ import qualified Juvix.Core.Pipeline as CorePipeline
 import qualified Juvix.Core.Translate as Translate
 import qualified Juvix.FrontendContextualise.InfixPrecedence.Environment as FE
 import Juvix.Library
+import qualified Juvix.Library.Feedback as Feedback
 import qualified Juvix.Library.Usage as Usage
 import qualified Juvix.Pipeline as Pipeline
 import qualified Michelson.Untyped as Untyped
 import Options
+import qualified System.IO.Temp as Temp
 import Types
+import qualified Prelude as P
 
-parse :: FilePath -> IO FE.FinalContext
-parse fin = do
-  core <- Pipeline.toCore ["stdlib/Prelude.ju", "stdlib/Michelson.ju", "stdlib/MichelsonAlias.ju", fin]
+type Code = Text
+
+type OutputCode = Text
+
+type Message = [P.String]
+
+type Pipeline = Feedback.FeedbackT Message IO
+
+-- | Function that parses code gives as input.
+parse :: Code -> Pipeline FE.FinalContext
+parse code = do
+  core <- liftIO $ toCore_wrap code
   case core of
-    Right ctx -> do
-      pure ctx
-    Left err -> do
-      T.putStrLn (show err)
-      exitFailure
+    Right ctx -> return ctx
+    Left err -> Feedback.FeedbackT $ return $ Feedback.Fail [show err]
+  where
+    toCore_wrap :: Code -> IO (Either Pipeline.Error FE.FinalContext)
+    toCore_wrap code = do
+      fp <- Temp.writeSystemTempFile "juvix-toCore.ju" (Text.unpack code)
```

Comment:
Will you be able to write to "juvix-toCore.ju" if you run this code twice (i.e. the file already exists)?

Reply:
Note this solution is just a workaround to be able to pass data into the 'old' pipeline, by means of a file. Ideally these functions will take actual code instead of a filepath.

The writeSystemTempFile function takes into account the file already existing by adding a process id and possible a number.